specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2023-06-28-preview/managedccf.json

```diff
@@ -0,0 +1,689 @@
+          "$ref": "#/definitions/LanguageRuntime"
+        },
+        "appSourceUri": {
+          "type": "string",
+          "readOnly": false,
+          "description": "Source Uri containing ManagedCCF code"
+        }
+      }
+    },
+    "CertificateTags": {
+      "description": "Tags for Managed CCF Certificates",
+      "type": "object",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "description": "Additional tags for Managed CCF Certificates",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "MemberIdentityCertificate": {
+      "type": "object",
+      "description": "Object representing MemberIdentityCertificate for Managed CCF.",
+      "title": "MemberIdentityCertificate",
+      "properties": {
+        "certificate": {
+          "type": "string",
+          "readOnly": false,
+          "description": "Member Identity Certificate"
+        },
+        "encryptionkey": {
+          "type": "string",
+          "readOnly": false,
+          "description": "Member Identity Certificate Encryption Key"
+        },
+        "tags": {
+          "items": {
+            "$ref": "#/definitions/CertificateTags"
+          }
+        }
+      }
+    },
+    "NodeCount": {
+      "type": "integer",
+      "format": "int32",
+      "description": "Number of CCF nodes in the Managed CCF.",
+      "title": "NodeCount",
+      "default": 3
+    },
+    "ManagedCCFProperties": {
+      "type": "object",
+      "description": "Additional Managed CCF properties.",
+      "title": "ManagedCCFProperties",
+      "properties": {
+        "appName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Unique name for the Managed CCF."
+        },
+        "appUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Endpoint for calling Managed CCF Service."
+        },
+        "identityServiceUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Endpoint for accessing network identity."
+        },
+        "memberIdentityCertificates": {
+          "type": "array",
+          "description": "List of member identity certificates for  Managed CCF",
+          "items": {
+            "$ref": "#/definitions/MemberIdentityCertificate"
+          },
+          "x-ms-identifiers": []
+        },
+        "deploymentType": {
+          "description": "Deployment Type of Managed CCF",
+          "$ref": "#/definitions/DeploymentType"
+        },
+        "runningState": {
+          "description": "Object representing RunningState for Managed CCF.",
+          "$ref": "#/definitions/RunningState"
+        },
+        "provisioningState": {
+          "description": "Provisioning state of Managed CCF Resource",
+          "readOnly": true,
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "nodeCount": {
```

Comment:
>nodeCount [](http://example.com/codeflow?start=9&length=9)

Maybe its too late now but... this might also have been something you could have modeled as a `.capacity` in the sku? Do you have skus?

Reply:
We are currently working on adding SKUs and plan on including this node count as part of different SKUs.

Reply:
If you want to support senarios autoscale, you probably want to be modelling the nodecount as an explicitly read/writable number property in the sku, then if your sku needs other properties is up to your pricing model and whether you have e.g. feature differentiation like "Premium SKU has X", or additional business logics you want to trigger, like "Premium SKU with high SLA requires at least 3 nodes" or whatever